File: src/Graph.py
import abc


class Graph(abc.ABC):
    @abc.abstractmethod
    def add_vertex(self, v):
        pass

    @abc.abstractmethod
    def add_edge(self, v, w):
        pass

    @abc.abstractmethod
    def delete_vertex(self, i):
        pass

    @abc.abstractmethod
    def delete_edge(self, v, w):
        pass

    @abc.abstractmethod
    def edge_exists(self, v, w):
        pass

    @abc.abstractmethod
    def order(self):
        pass

    @abc.abstractmethod
    def edge_amount(self):
        pass

    @abc.abstractmethod
    def get_vertex(self, i):
        pass

    @abc.abstractmethod
    def get_adjacent_list(self, v):
        pass

File: src/src/Graph.java
import java.util.List;

public interface Graph {
    class Edge {
        int from;
        int to;

        Edge(int from, int to) {
            this.from = from;
            this.to = to;
        }
    }

    void addVertex(String v);

    void addEdge(Edge e);

    void deleteVertex(String v);

    void deleteEdge(Edge e);

    boolean edgeExists(int v, int w);

    List<Integer> getAdjacentList(int v);
}
